Breeze.TumbleBit.Client/TumbleBitManager.cs

```diff
@@ -336,46 +336,78 @@ public enum TumbleState
             // - If this is a first connection, this.TumblerAddress will be null
             // - If we were previously connected to a server, its URI would have been stored in the
             //   tumbling_state.json, and will have been loaded into this.TumblerAddress already
-
-            // If the -ppuri command line option wasn't used to bypass the registration store lookup
             if (this.TumblerAddress == null)
             {
                 List<RegistrationRecord> registrations = this.registrationStore.GetAll();
 
                 if (registrations.Count < MINIMUM_MASTERNODE_COUNT)
                 {
-                    this.logger.LogDebug("Not enough masternode registrations downloaded yet: " + registrations.Count);
+                    this.logger.LogDebug($"Not enough masternode registrations downloaded yet: {registrations.Count}");
                     return Result.Fail<ClassicTumblerParameters>("Not enough masternode registrations downloaded yet");
                 }
-                
+
                 registrations.Shuffle();
 
-                // Since the list is shuffled, we can simply iterate through it and try each server until one is valid & reachable
+                // Since the list is shuffled, we can simply iterate through it and try each server until one is valid & reachable.
                 foreach (RegistrationRecord record in registrations)
                 {
-                    this.TumblerAddress = "ctb://" + record.Record.OnionAddress + ".onion?h=" + record.Record.ConfigurationHash;
+                    this.TumblerAddress = $"ctb://{record.Record.OnionAddress}.onion?h={record.Record.ConfigurationHash}";
+
+                    //Do not attempt a connection to the Masternode which is blacklisted
+                    if (masternodeBlacklist != null && masternodeBlacklist.Contains(this.TumblerAddress)) {
+                        this.logger.LogDebug($"Skipping connection attempt to blacklisted masternode {this.TumblerAddress}");
+                        continue;
+                    }
 
-                    var attemptConnection = await TryUseServer();
+                    try
+                    {
+                        var tumblerParameterResult = await TryUseServer();
 
-                    if (!attemptConnection.Failure)
+                        if (tumblerParameterResult.Success)
+                        {
+                            return tumblerParameterResult;
+                        }
+                    }
+                    catch (Exception e)
                     {
-                        return attemptConnection;
+                        this.logger.LogDebug(e, $"Unable to connect to masternode: {this.TumblerAddress}");
                     }
                 }
 
-                // If we reach this point, no servers were reachable
-                this.logger.LogDebug("Did not find a valid registration");
+                // If we reach this point, no servers were reachable.

```

Comment:
Is this comment redundant as we can see the same information in the LogDebug comment below ?